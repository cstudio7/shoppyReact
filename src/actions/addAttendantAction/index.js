import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addAttendantRequest';

const { ADD_ATTENDANT_FAILURE, ADD_ATTENDANT_LOADING, ADD_ATTENDANT_SUCCESS } = actionTypes;

const addAttendantLoading = type => ({
  type
});

const addAttendantSuccess = payload => ({
  type: ADD_ATTENDANT_SUCCESS,
  payload
});

const addAttendantFailure = payload => ({
  type: ADD_ATTENDANT_FAILURE,
  payload
});

const addStoreAttendant = attendantDetails => async (dispatch) => {
  try {
    dispatch(addAttendantLoading(ADD_ATTENDANT_LOADING));
    const response = await request.addAttendantRequest(attendantDetails);
    dispatch(addAttendantSuccess(response.data.rows));
    toast.success('Attendant created successfully');
  } catch (error) {
    dispatch(addAttendantFailure(error.response.data.message));
  }
};

export { addStoreAttendant, addAttendantLoading, addAttendantSuccess, addAttendantFailure };
