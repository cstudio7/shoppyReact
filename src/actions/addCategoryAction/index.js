import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addCategoryRequest';

const { ADD_CATEGORY_FAILURE, ADD_CATEGORY_LOADING, ADD_CATEGORY_SUCCESS } = actionTypes;

const addCategoryLoading = type => ({
  type
});

const addCategorySuccess = payload => ({
  type: ADD_CATEGORY_SUCCESS,
  payload
});

const addCategoryFailure = payload => ({
  type: ADD_CATEGORY_FAILURE,
  payload
});

const addCategory = categoryDetails => async (dispatch) => {
  try {
    dispatch(addCategoryLoading(ADD_CATEGORY_LOADING));
    const response = await request.addCategoryRequest(categoryDetails);
    dispatch(addCategorySuccess(response.data.rows));
    toast.success('Category created successfully');
  } catch (error) {
    dispatch(addCategoryFailure(error.response.data));
  }
};

export { addCategory, addCategoryLoading, addCategorySuccess, addCategoryFailure };
