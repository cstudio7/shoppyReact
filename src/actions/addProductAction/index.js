import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addProductRequest';

const { ADD_PRODUCT_FAILURE, ADD_PRODUCT_LOADING, ADD_PRODUCT_SUCCESS } = actionTypes;

const addProductLoading = type => ({
  type
});

const addProductSuccess = payload => ({
  type: ADD_PRODUCT_SUCCESS,
  payload
});

const addProductFailure = payload => ({
  type: ADD_PRODUCT_FAILURE,
  payload
});

const addProduct = (productDetails, categoryId) => async (dispatch) => {
  try {
    dispatch(addProductLoading(ADD_PRODUCT_LOADING));
    const response = await request.addProductRequest(productDetails, categoryId);
    dispatch(addProductSuccess(response.data.rows));
    toast.success('Product created successfully');
  } catch (error) {
    dispatch(addProductFailure(error.response.data));
  }
};

export { addProduct, addProductLoading, addProductSuccess, addProductFailure };
