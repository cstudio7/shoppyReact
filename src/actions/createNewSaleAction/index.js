import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/createSaleRequest';

const { CREATE_SALE_FAILURE, CREATE_SALE_LOADING, CREATE_SALE_SUCCESS } = actionTypes;

const createSaleLoading = type => ({
  type
});

const createSaleSuccess = payload => ({
  type: CREATE_SALE_SUCCESS,
  payload
});

const createSaleFailure = payload => ({
  type: CREATE_SALE_FAILURE,
  payload
});

const createNewSale = (salesArray, price) => async (dispatch) => {
  try {
    dispatch(createSaleLoading(CREATE_SALE_LOADING));
    const response = await request.createSaleRequest(salesArray, price);
    dispatch(createSaleSuccess(response.data.resultArray));
    toast.success('You just sold these items, keep it up');
  } catch (error) {
    dispatch(createSaleFailure(error.response.data.error));
  }
};

export { createNewSale, createSaleLoading, createSaleSuccess, createSaleFailure };
