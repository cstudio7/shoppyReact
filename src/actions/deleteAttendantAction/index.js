import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/deleteAttendantRequest';

const {
  DELETE_ATTENDANT_FAILURE,
  DELETE_ATTENDANT_LOADING,
  DELETE_ATTENDANT_SUCCESS
} = actionTypes;

const deleteAttendantLoading = type => ({
  type
});

const deleteAttendantSuccess = payload => ({
  type: DELETE_ATTENDANT_SUCCESS,
  payload
});

const deleteAttendantFailure = payload => ({
  type: DELETE_ATTENDANT_FAILURE,
  payload
});

const deleteStoreAttendant = attendantId => async (dispatch) => {
  try {
    dispatch(deleteAttendantLoading(DELETE_ATTENDANT_LOADING));
    const response = await request.deleteAttendantRequest(attendantId);
    dispatch(deleteAttendantSuccess(response.data.Message));
  } catch (error) {
    dispatch(deleteAttendantFailure(error.response.data.message));
  }
};

export {
  deleteStoreAttendant,
  deleteAttendantLoading,
  deleteAttendantSuccess,
  deleteAttendantFailure
};
