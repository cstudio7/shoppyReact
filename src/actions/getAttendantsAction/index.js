import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/getAttendantsRequest';

const { GET_ATTENDANTS_FAILURE, GET_ATTENDANTS_LOADING, GET_ATTENDANTS_SUCCESS } = actionTypes;

const attendantsLoading = type => ({
  type
});

const attendantsSuccess = payload => ({
  type: GET_ATTENDANTS_SUCCESS,
  payload
});

const attendantsFailure = payload => ({
  type: GET_ATTENDANTS_FAILURE,
  payload
});

const storeAttendants = () => async (dispatch) => {
  try {
    dispatch(attendantsLoading(GET_ATTENDANTS_LOADING));
    const response = await request.getAttendantsRequest();
    dispatch(attendantsSuccess(response.data.rows));
  } catch (error) {
    dispatch(attendantsFailure(error.response.data.Message));
  }
};

export { storeAttendants, attendantsFailure, attendantsSuccess, attendantsLoading };
