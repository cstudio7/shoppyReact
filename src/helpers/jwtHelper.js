import jwtDecode from 'jwt-decode';

const authentication = 'Authentication';

const setToken = token => localStorage.setItem(authentication, token);

const getUserInfo = () => {
  const token = localStorage.getItem(authentication);
  try {
    const payLoad = jwtDecode(token);
    return payLoad;
  } catch (error) {
    return null;
  }
};

const getToken = localStorage.getItem(authentication);

export { setToken, getToken, getUserInfo };
