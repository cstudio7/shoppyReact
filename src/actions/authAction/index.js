import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/loginRequest';
import { setToken } from '../../helpers/jwtHelper';

const { USER_LOGIN_FAILURE, USER_LOGIN_LOADING, USER_LOGIN_SUCCESS } = actionTypes;

const loginLoading = type => ({
  type
});

const loginSuccess = payload => ({
  type: USER_LOGIN_SUCCESS,
  payload
});

const loginFailure = payload => ({
  type: USER_LOGIN_FAILURE,
  payload
});

const userLogin = userDetails => async (dispatch) => {
  try {
    dispatch(loginLoading(USER_LOGIN_LOADING));
    const response = await request.loginRequest(userDetails);
    setToken(response.data.token);
    dispatch(loginSuccess(response));
    toast.success(response.data.message);
  } catch (error) {
    dispatch(loginFailure(error.response));
    toast.error(error.response.data.message);
  }
};

export { userLogin, loginFailure, loginLoading, loginSuccess };
