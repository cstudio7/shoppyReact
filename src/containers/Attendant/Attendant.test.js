import React from 'react';
import { shallow } from 'enzyme';
import { AttendantPage } from './index';
import { storeAttendants } from '../../actions/getAttendantsAction';
import { getUserInfo } from '../../helpers/jwtHelper';

describe('Test the Attendant Component', () => {
  beforeAll(() => {
    localStorage.setItem(
      'Authentication',
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlMb2FkIjp7ImlkIjoiN2NhM2JjYzktZTMzOS00OTcwLTk0YjMtYjI5YTg1N2ViM2ExIiwicm9sZSI6W119LCJpYXQiOjE1NTE0NjE4OTcsImV4cCI6MTU1MTU0ODI5N30.IzGf0L_pH9W_gN2AC2Ee5aWqwSO4l3fa46ZrcyUbMKU'
    );
  });
  afterEach(() => {
    localStorage.removeItem('Authentication');
  });
  it('should match snapshot', () => {
    const allAttendants = {
      successResponse: [
        {
          id: '1',
          email: 's@mail.com',
          image_url: 'imageurl',
          mobile_number: '09085674565',
          first_name: 'welcome'
        }
      ]
    };
    const wrapper = shallow(
      <AttendantPage storeAttendants={storeAttendants} allAttendants={allAttendants} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test the Attendant Component', () => {
  beforeAll(() => {});
  afterEach(() => {
    localStorage.removeItem('Authentication');
  });
  it('should match snapshot', () => {
    const result = getUserInfo();
    const allAttendants = {
      successResponse: [
        {
          id: '1',
          email: 's@mail.com',
          image_url: 'imageurl',
          mobile_number: '09085674565',
          first_name: 'welcome'
        }
      ]
    };
    const wrapper = shallow(
      <AttendantPage storeAttendants={storeAttendants} allAttendants={allAttendants} />
    );
    expect(result).toBeNull();
    expect(wrapper).toMatchSnapshot();
  });
});
