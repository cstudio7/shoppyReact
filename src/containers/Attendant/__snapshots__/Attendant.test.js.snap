// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Attendant Component should match snapshot 1`] = `<Unauthorized />`;

exports[`Test the Attendant Component should match snapshot 2`] = `
<Redirect
  push={false}
  to="/login"
/>
`;
